src/back/spv/writer.rs

```diff
@@ -1208,6 +1208,75 @@ impl Writer {
         null_id
     }
 
+    fn write_constant_expr(
```

Comment:
I don't entirely understand what's happening here, but it seems to me that, because an expression in an arena may only refer to expressions that appear before it in the arena, and we process `const_expressions` in order, each expression may always assume that `constant_ids` has been populated for all its subexpressions, and thus no recursion should ever be necessary.

Reply:
The fn was previously called for each constant initializer instead. Will update.